assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -207,35 +438,178 @@ subroutine obs_inc_info(obs_kind, filter_kind, rectangular_quadrature, gaussian_
 ! Temporary approach for setting the details of how to assimilate this observation
 ! This example is designed to reproduce the squared forward operator results from paper
 
-
-! Set the observation increment details for each type of quantity
-if(obs_kind == QTY_STATE_VARIABLE) then
+!use default values if qcf_table_filename is not in namelist
+if (use_qty_defaults) then
    filter_kind = BOUNDED_NORMAL_RHF
    bounded_below = .false.;    bounded_above = .false.
    lower_bound   = missing_r8; upper_bound   = missing_r8
-elseif(obs_kind == QTY_TRACER_CONCENTRATION) then
-   filter_kind = BOUNDED_NORMAL_RHF
-   bounded_below = .true.; bounded_above = .false.
-   lower_bound   = 0.0_r8; upper_bound   = missing_r8
-elseif(obs_kind == QTY_TRACER_SOURCE) then
-   filter_kind = BOUNDED_NORMAL_RHF
-   bounded_below = .true.; bounded_above = .false.
-   lower_bound   = 0.0_r8; upper_bound   = missing_r8
-else
-   write(*, *) 'Illegal obs_kind in obs_error_info'
-   stop
+   spread_restoration = .false.
+   return
 endif
 
-! Default settings for now for Icepack and tracer model tests
-sort_obs_inc = .false.
-spread_restoration = .false.
+!get actual name of QTY from integer index
+qty_name = get_name_for_quantity(obs_qty)
+
+!find location of QTY in qcf_table_data structure
+QTY_loc = findloc(specified_qtys, qty_name)
+
+if (QTY_loc(1) == 0) then
+   !use default values if QTY is not in table
+   filter_kind = BOUNDED_NORMAL_RHF
+   bounded_below = .false.;    bounded_above = .false.
+   lower_bound   = missing_r8; upper_bound   = missing_r8
+   spread_restoration = .false.
```

Comment:
You have a couple of places where the default is set, here and line 442.
It might be good to set the default values in one place.

Reply:
This comment can be addressed in a future refactor/pull request